pkg/providers/tinkerbell/comparator.go

```diff
@@ -0,0 +1,51 @@
+package tinkerbell
+
+import (
+	clusterv1 "sigs.k8s.io/cluster-api/api/v1beta1"
+	controlplanev1 "sigs.k8s.io/cluster-api/controlplane/kubeadm/api/v1beta1"
+)
+
+// VersionComparable enables kubernetes version retrieval.
+type VersionComparable interface {
```

Comment:
I think we should [stick with Go idioms](https://go.dev/doc/effective_go#interface-names) in this instance. I'm not sold on the use of `Comparable` because it really doesn't add anything to the type vs `Versionable`.

Reply:
Updated naming to `Versionable`